core/src/test/java/nl/nn/adapterframework/filesystem/FileSystemActorTest.java

```diff
@@ -1167,7 +1167,39 @@ public void fileSystemActorRmdirActionTest() throws Exception {
 		// test
 		assertFalse("Expected folder [" + folder + "] " + "not to be present", actual);
 	}
+	@Test
+	public void fileSystemActorRmNonEmptyDirActionTest() throws Exception {
```

Comment:
You should also test that if removeNonEmptyDirecty=false, that an exception is thrown if the folder is not empty.